trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/remote/session/RemoteSessionImpl.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.s3.proxy.server.remote.session;
+
+import io.trino.s3.proxy.server.credentials.Credential;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentials;
+import software.amazon.awssdk.auth.credentials.AwsSessionCredentials;
+import software.amazon.awssdk.services.sts.StsClient;
+import software.amazon.awssdk.services.sts.auth.StsAssumeRoleCredentialsProvider;
+
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+class RemoteSessionImpl
+        implements RemoteSession
+{
+    private final StsAssumeRoleCredentialsProvider credentialsProvider;
```

Comment:
should we make this a `AwsCredentailProvider`? `StsAssumeRoleCredentialProvider` implements that and provides the `resolveCredentials()` function
this class is pretty agnostic already to what provides the credentials

Reply:
See change below